### PR TITLE
feat(Maliki): add new Presence

### PR DESCRIPTION
**Pull request by RisingSunLight42**
## Description 

Adding a new presence, I've followed the advices on the PR #6366 :
- using the path of the url instead of `document.URL`
- use of more dynamics informations and add of few presence's settings

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

proof_screen_1_6
proof_screen_1_5
proof_screen_1_4
proof_screen_1_3
proof_screen_1_2
proof_screen_1_1

</details>

### CODE REVIEW

File: websites/M/Maliki/iframe.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const iframe = new iFrame();
+
+iframe.on("UpdateData", async () => {
+	if (document.querySelector(".artiste") && document.querySelector(".titre")) {
```
Comment (ImRodry) [inline websites/M/Maliki/iframe.ts:4]: make these constants to avoid computing them twice

File: websites/M/Maliki/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.7",
+	"author": {
+		"name": "RisingSunLight",
+		"id": "240521747852558347"
+	},
+	"service": "Maliki",
+	"description": {
+		"en": "Presence which helps you to follow your activity on the official website of Maliki !",
+		"fr": "Présence qui permet de suivre votre activité sur le site officiel de Maliki !"
```
Comment (ImRodry) [inline websites/M/Maliki/dist/metadata.json:10]: these are descriptions of the website, not the presence...
```suggestion
		"en": "I hate rambling about myself... Apart from in my comics, my novels and on my blog that is!",
		"fr": "J'ai horreur de parler de moi... Ailleurs que dans mes BD, mes romans et sur mon blog !",
		"es": "Odio hablar de mi... Aparte de mis cómics, mis novelas y en mi blog!"
```

File: websites/M/Maliki/dist/metadata.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	},
+	"url": [
+		"maliki.com",
+		"malimode.maliki.com"
+	],
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/QDOaEbp.jpg",
```
Comment (ImRodry) [inline websites/M/Maliki/dist/metadata.json:17]: must be 512x512

File: websites/M/Maliki/dist/metadata.json [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+	"thumbnail": "https://i.imgur.com/LDOrHM3.jpg",
```
Comment (ImRodry) [inline websites/M/Maliki/dist/metadata.json:18]: if you can make the resolution of this one a bit better that would be nice

File: websites/M/Maliki/presence.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
+const presence = new Presence({
+		clientId: "990204576189808670",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+interface songPlayingInformations {
```
Comment (ImRodry) [inline websites/M/Maliki/presence.ts:6]:
```suggestion
interface SongPlayingData {
```

File: websites/M/Maliki/presence.ts [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+let songInfo: songPlayingInformations = null;
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey:
+				document.location.hostname === "malimode.maliki.com"
+					? "malimode"
+					: "maliki_1024",
+			startTimestamp: browsingTimestamp,
+			buttons: [{ label: "Voir le site", url: "https://maliki.com/" }],
```
Comment (ImRodry) [inline websites/M/Maliki/presence.ts:20]: we don't allow buttons to the main page

File: websites/M/Maliki/presence.ts [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+		},
+		//* Get options of the presence
+		[privacy, time] = await Promise.all([
+			presence.getSetting<boolean>("privacy"),
+			presence.getSetting<boolean>("time"),
+		]);
+	if (!time) delete presenceData.startTimestamp; // delete the startTimestamp if time option desactivated
```
Comment (ImRodry) [inline websites/M/Maliki/presence.ts:27]: we don't need a comment on that one

File: websites/M/Maliki/presence.ts [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey:
+				document.location.hostname === "malimode.maliki.com"
+					? "malimode"
+					: "maliki_1024",
+			startTimestamp: browsingTimestamp,
+			buttons: [{ label: "Voir le site", url: "https://maliki.com/" }],
+		},
+		//* Get options of the presence
```
Comment (ImRodry) [inline websites/M/Maliki/presence.ts:22]: no need to have this either

File: websites/M/Maliki/presence.ts [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+		[privacy, time] = await Promise.all([
+			presence.getSetting<boolean>("privacy"),
+			presence.getSetting<boolean>("time"),
+		]);
+	if (!time) delete presenceData.startTimestamp; // delete the startTimestamp if time option desactivated
+
+	//* Change the detail depending on the current URL
```
Comment (ImRodry) [inline websites/M/Maliki/presence.ts:29]: this isn't needed

File: websites/M/Maliki/presence.ts [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+			presenceData.details = "Partage son personnage";
+			presenceData.largeImageKey = "malimode";
+			break;
+		default: {
+			if (document.location.pathname.includes("/strips/")) {
+				// If we are on a strip page, modify the details
+				presenceData.details = privacy ? "Lit un strip" : "Lit le strip :";
+				presenceData.state = privacy
+					? ""
+					: document.querySelector(".singleTitle").textContent;
```
Comment (ImRodry) [inline websites/M/Maliki/presence.ts:76]:
```suggestion
				if (!privacy) presenceData.state =
					document.querySelector(".singleTitle").textContent;
```

File: websites/M/Maliki/presence.ts [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+			}
+			if (document.location.pathname.includes("/bonus/")) {
+				// If we are on a bonus page, modify the details
+				presenceData.details = privacy
+					? "Regarde un bonus"
+					: "Regarde le bonus :";
+				presenceData.state = privacy
+					? ""
+					: document.querySelector(".singleTitle").textContent;
```
Comment (ImRodry) [inline websites/M/Maliki/presence.ts:85]: ditto

File: websites/M/Maliki/presence.ts [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+			}
+			break;
+		}
+	}
+
+	if (presenceData.details) presence.setActivity(presenceData);
+	else presence.setActivity();
+});
+
+presence.on("iFrameData", async (data: songPlayingInformations) => {
```
Comment (ImRodry) [inline websites/M/Maliki/presence.ts:108]: no need for this to be async

**Review by ImRodry — CHANGES_REQUESTED** [review]
@Kozou4ever please review the strings on this PR

File: websites/M/Maliki/dist/metadata.json (hunk 2 above)
Comment (RisingSunLight42) [reply websites/M/Maliki/dist/metadata.json:10]: Oups, okay, I've didn't understand this in that way!

File: websites/M/Maliki/iframe.ts (hunk 1 above)
Comment (RisingSunLight42) [reply websites/M/Maliki/iframe.ts]: Done in commit 21851275d7a887dfd0a995e019632c902ce63e77

File: websites/M/Maliki/presence.ts (hunk 12 above)
Comment (RisingSunLight42) [reply websites/M/Maliki/presence.ts]: Corrected in 478d0d23d9b6c22f270fba74fa3b89e24676d745

File: websites/M/Maliki/presence.ts (hunk 11 above)
Comment (RisingSunLight42) [reply websites/M/Maliki/presence.ts]: Corrected in 478d0d23d9b6c22f270fba74fa3b89e24676d745

File: websites/M/Maliki/presence.ts (hunk 10 above)
Comment (RisingSunLight42) [reply websites/M/Maliki/presence.ts]: Corrected in 478d0d23d9b6c22f270fba74fa3b89e24676d745

File: websites/M/Maliki/presence.ts (hunk 9 above)
Comment (RisingSunLight42) [reply websites/M/Maliki/presence.ts]: Corrected in 478d0d23d9b6c22f270fba74fa3b89e24676d745

File: websites/M/Maliki/presence.ts (hunk 8 above)
Comment (RisingSunLight42) [reply websites/M/Maliki/presence.ts]: Corrected in 478d0d23d9b6c22f270fba74fa3b89e24676d745

File: websites/M/Maliki/presence.ts (hunk 7 above)
Comment (RisingSunLight42) [reply websites/M/Maliki/presence.ts]: Corrected in 478d0d23d9b6c22f270fba74fa3b89e24676d745

File: websites/M/Maliki/presence.ts (hunk 6 above)
Comment (RisingSunLight42) [reply websites/M/Maliki/presence.ts]: Removed in 478d0d23d9b6c22f270fba74fa3b89e24676d745
I'll add buttons redirecting to subpages when the user is on it.

File: websites/M/Maliki/presence.ts (hunk 5 above)
Comment (RisingSunLight42) [reply websites/M/Maliki/presence.ts]: Rename in 478d0d23d9b6c22f270fba74fa3b89e24676d745

File: websites/M/Maliki/dist/metadata.json (hunk 3 above)
Comment (RisingSunLight42) [reply websites/M/Maliki/dist/metadata.json:18]: I've changed the size on imgur, this should be ok.

File: websites/M/Maliki/dist/metadata.json (hunk 4 above)
Comment (RisingSunLight42) [reply websites/M/Maliki/dist/metadata.json]: I've made it better, idk if it's ok, but it's the best I can do 5340f44d9f71338d887645989a3c948545bcb49b

File: websites/M/Maliki/presence.ts (hunk 5 above)
Comment (ImRodry) [reply websites/M/Maliki/presence.ts]: I changed the casing on purpose, it should use the name I suggested.

Comment (RisingSunLight42) [reply websites/M/Maliki/presence.ts]: I don't undertstand what you mean. I've changed the name to the one who've suggested ?

Comment (RisingSunLight42) [reply websites/M/Maliki/presence.ts]: Oh okay, I'm dumb, i've didn't see the uppercase S

Comment (RisingSunLight42) [reply websites/M/Maliki/presence.ts]: Right rename in a65fb62f8172dd2e9827123bae9036b2484f372c

**Review by RisingSunLight42 — COMMENTED** [review]
I've done everything requested. I don't know if I can approve all the changes myself ?

**Review by ImRodry — APPROVED** [review]
LGTM, waiting on language review

File: websites/M/Maliki/presence.ts [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+					: "Regarde la page principale";
+			break;
+		case "/strips/":
+			presenceData.details = "Parcourt les strips";
+			break;
+		case "/actualites/":
+			presenceData.details = "Parcourt les actualités";
+			break;
+		case "/bonus/":
+			presenceData.details = "Parcours les bonus";
```
Comment (Kozou4ever) [inline websites/M/Maliki/presence.ts:41]: Salut, c'est un peu incohérent ici, soit tu mets à la deuxième personne soit à la troisième personne du singulier... Je te conseille d'ailleurs de tout mettre à la deuxième personne dans la presence :)

Comment (Kozou4ever) [reply websites/M/Maliki/presence.ts:41]: j'ai dis deuxième personne mais c'est plus la première ^^

Comment (RisingSunLight42) [reply websites/M/Maliki/presence.ts:41]: Ok ! Merci de la remarque, je corrige ça asap

Comment (RisingSunLight42) [reply websites/M/Maliki/presence.ts]: Fait ici 894e1760d059b24b05a966d23785ecd4f5149142

**Review by itsmelouis — APPROVED** [review]
LGTM as regards translations

**Review by EncryptedDev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**